### PR TITLE
Enforce `Bytes` and remove obsolete hex-utils

### LINKED ISSUE #395: Investigate use of `array-bytes` or `Bytes` instead of self implemented `from_hex`
https://github.com/hack-ink/array-bytes/blob/main/src/lib.rs should do the job I think.

### PR DESCRIPTION
**Pull request by haerdib**
- The opaque functions now enforce `Bytes` instead of `Vec<u8>`. `Bytes` implements serialization by default with `hex_encode`, rendering the hex utils as obsolete.
- Removed utils and moved the only, not obsolete function `storage_key(..)` to `lib.rs`.

- [x] merge PR #421 

closes #395

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -23,9 +23,18 @@ pub use ac_compose_macros::*;
 pub use ac_node_api::*;
 pub use ac_primitives::*;
 pub use api::*;
-pub use utils::*;
 
 pub mod api;
 pub mod extrinsic;
 pub mod rpc;
-pub mod utils;
```
Comment (Niederb) [inline src/lib.rs:31]: I'm wondering if the tag `E1-breaksnothing` is accurate given that we remove the `utils` module. Couldn't this affect some users?

**Review by Niederb — COMMENTED** [review]
Other than the question regarding `E1-breaksnothing` this looks good to me.

File: src/lib.rs (hunk 1 above)
Comment (haerdib) [reply src/lib.rs:31]: Agreed, it does break the api. Will correct this

Comment (haerdib) [reply src/lib.rs:31]: Thanks for pointing it out!

**Review by Niederb — APPROVED** [review]

**Pull request merged**